sphinx/_cli/util/errors.py

```diff
@@ -225,9 +225,11 @@ def print_red(*values: str) -> None:
 
     print_red(__('Exception occurred:'))
     print_err(formatted_tb)
-    traceback_info_path = save_traceback(
+    traceback_output = full_exception_context(
         exception, message_log=message_log, extensions=extensions
     )
+    print_err(traceback_output)
+    traceback_info_path = write_temporary_file(traceback_output)
```

Comment:
Should we move this earlier? Currently it only executes on the general case of `Exception` after the specific exceptions have been handled before it.

Reply:
As you wish. My thinking was to print in the log as soon as we have the traceback, then write on the disk, as there is a slightly greater chance that writing on the disk might failed compared to the failing to print in the log. But I might over think things. So I'm going to just applied your suggestion.

Reply:
Sorry, I didn't mean changing the order of print vs saving to disk, but moving this entire chunk upwards -- see the rest of the function for where various other error types are handled. What do you think?

Reply:
Oh OK. So the idea would be to move the unconditional traceback printing before the handling of each specific kind of exception, each of which can interrupt the execution of the function with a `return`.

At that stage, I just wonder if there is any point in handling each exception type, since the `traceback_output` already does the job of providing all the contextual information.

So I update the PR to show you what my proposal would look like, simplifying the code and displaying the detailed traceback if any exception occur. If you don't like it I'll revert that change.